tests/test_processing_stack.py

```diff
@@ -0,0 +1,117 @@
+import logging
+import os
+import time
+
+from pytest import mark
+
+APPLICATION_TAG_KEY = "ApplicationName"
+APPLICATION_TAG_VAL = "geospatial-data-lake"
+
+ENVIRONMENT_TAG_KEY = "EnvironmentType"
+ENVIRONMENT_TAG_VAL = os.environ.get("DEPLOY_ENV", "dev")
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def detect_compute_environment(batch_client):
+    compute_environment_detection_response = batch_client.describe_compute_environments()
+    logger.info("Detected Compute Environments: %s", compute_environment_detection_response)
+
+    for compute_environment in compute_environment_detection_response["computeEnvironments"]:
+        if (
+            compute_environment["tags"][ENVIRONMENT_TAG_KEY] == ENVIRONMENT_TAG_VAL
+            and compute_environment["tags"].get(APPLICATION_TAG_KEY, None) == APPLICATION_TAG_VAL
+        ):
+            datalake_compute_environment = compute_environment
+            logger.info("Datalake Compute Environment: %s", datalake_compute_environment)
+            break
+
+    return datalake_compute_environment
+
+
+def launch_compute_environment_resources(batch_client, datalake_compute_environment):
+    """
+    Trigger immediate launch of computing resources. This will significantly speed up Batch jobs
+    execution.
+    """
+    if (
+        int(datalake_compute_environment["computeResources"]["minvCpus"]) < 1
+        or int(datalake_compute_environment["computeResources"]["desiredvCpus"]) < 1
+    ):
+        batch_client.update_compute_environment(
+            computeEnvironment=datalake_compute_environment["computeEnvironmentName"],
+            computeResources={"minvCpus": 1, "desiredvCpus": 1},
+        )
+
+
+def detect_state_machine(stepfunctions_client):
+    state_machines_detection_response = stepfunctions_client.list_state_machines()
+    logger.info("Detected State Machines: %s", state_machines_detection_response)
+
+    for state_machine in state_machines_detection_response["stateMachines"]:
+
+        tags_detection_response = stepfunctions_client.list_tags_for_resource(
+            resourceArn=state_machine["stateMachineArn"]
+        )
+
+        # reformat returned array of tags to dictionary
+        state_machine_tags = {}
+        for tag in tags_detection_response["tags"]:
+            state_machine_tags[tag["key"]] = tag["value"]
+
+        if (
+            state_machine_tags[ENVIRONMENT_TAG_KEY] == ENVIRONMENT_TAG_VAL
+            and state_machine_tags.get(APPLICATION_TAG_KEY, None) == APPLICATION_TAG_VAL
+        ):
+            datalake_state_machine = state_machine
+            logger.info("Datalake State Machine: %s", datalake_state_machine)
+            break
```

Comment:
We could simplify the last three lines to just

```
            logger.info("Datalake State Machine: %s", state_machine)
            return state_machine
```

This has the added advantage that we won't ever return a non-existing variable.

We should also do something like `raise NoStateMachineFound(APPLICATION_TAG_VAL, ENVIRONMENT_TAG_VAL)` as the last line outside the loop to avoid returning `None` and kicking the ball further down the road.

Reply:
I get Pylint warning "inconsistent-return-statements" if I return from condition.

Reply:
Did you change it to `raise` t the end of the function as well? Because then there's only one `return` statement.

Reply:
Sorry, what I should change to `raise` ?

Reply:
If we reach the end of the method without finding a match, then I would expect something like `raise NoStateMachineFound(APPLICATION_TAG_VAL, ENVIRONMENT_TAG_VAL)` as the last line.

Reply:
@l0b0 , are you able to comment on this please ? Thanks

Reply:
Done in 7f60bdd1c351d8dab432c00ec6830644a6707b2e